app/models/project.rb

```diff
@@ -127,431 +104,22 @@ def user_role(user)
     return (self.user_projects.select { |up| up.user == user }).first.role
   end
 
-  def modules_without_group
-    MyModule.where(project_id: id).where(my_module_group: nil)
-      .where(archived: false)
-  end
-
-  def active_module_groups
-    self.my_module_groups.joins(:my_modules)
-      .where('my_modules.archived = ?', false)
-      .distinct
-  end
-
-  def assigned_samples
-    Sample.joins(:my_modules).where(my_modules: {id: my_modules} )
-  end
-
-  def unassigned_samples(assigned_samples)
-    Sample.where(organization_id: organization).where.not(id: assigned_samples)
+  def project_my_modules
```

Comment:
Urm. Project has many experiments. Each experiment has many modules. So wouldn't the correct relation be more like:
```ruby
MyModule.where('experiment_id IN (?)', experiments.select(:id))
```